src/pipelinedefinition.cpp

```diff
@@ -85,14 +87,66 @@ Status PipelineDefinition::validate(ModelManager& manager) {
     return validationResult;
 }
 
+Status PipelineDefinition::initializeNodeResources() {
+    for (const auto& nodeInfo : nodeInfos) {
+        if (nodeInfo.kind == NodeKind::CUSTOM) {
+            void* customNodeLibraryInternalManager = nullptr;
+            auto it = nodeResources.find(nodeInfo.nodeName);
+            bool internalManagerExists = (it != nodeResources.end());
+            if (internalManagerExists) {
+                customNodeLibraryInternalManager = nodeResources.at(nodeInfo.nodeName);
+            }
+            auto params = createCustomNodeParamArray(nodeInfo.parameters).get();
+            if (!nodeInfo.library.isValid()) {
+                SPDLOG_LOGGER_ERROR(modelmanager_logger, "Pipeline: {} node: {} refers to incorrect library", pipelineName, nodeInfo.nodeName);
```

Comment:
Incorrect might indicate that it refers to existing, valid library but not suitable for this pipeline. I think "invalid" or "corrupted" would fit better